### ISSUE TITLE
relation "XXX.aws_securityhub_finding" does not exist at character Y

### ISSUE THREAD

**Issue by captainfalcon23** [turbot/steampipe#3486]
Hi,

I am getting a sudden error with a steampipe query which I have been running for 6+ months without issue (started occurring around 2 weeks ago). Some context - I have around 15 Gitlab jobs which use some standard scripts I've written to get some data (generally from AWS) and create a CSV. Each time these run, they install the latest version of the AWS, CSV and Steampipe plugins, and latest version of Steampipe itself. This is all executed within a docker runner.

Of all of these 15+ jobs, only 1 is suddenly failing. This is a completely stripped back query which is failing:
```
su steampipe -c 'steampipe query "select region as \"Region\" from XXX.aws_securityhub_finding limit 1;" --output csv' > output.1
```

The following errors are reported in the logs:
```
2023-05-29 05:56:16.419 UTC [TRACE] steampipe: updated the required search path to "public","internal"
2023-05-29 05:56:16.419 UTC [TRACE] steampipe: session search path is already correct - nothing to do
Warning: executeQueries: query 1 of 1 failed: ERROR: relation "XXX.aws_securityhub_finding" does not exist (SQLSTATE 42P01)
2023-05-29 05:56:16.420 UTC [TRACE] steampipe: close local client 0xc0006dbaa0
2023-05-29 05:56:16.420 UTC [TRACE] steampipe: local client not NIL
2023-05-29 05:56:16.420 UTC [TRACE] steampipe: DbClient.Close &{1 0 0 0xc0008e52c0 0xc000091560 <nil> 0x15d6200 <nil> <nil> 1 1 600000000000 0 60000000000 60000000000 0xc0009fad80 {0 {0 0}} 0xc0009fade0}
2023-05-29 05:56:16.420 UTC [TRACE] steampipe: local client close complete
2023-05-29 05:56:16.420 UTC [TRACE] steampipe: shutdown local service query
```

I've tried hard coding an older version of the AWS plugin (0.98.0) and old version of Steampipe (v0.19.0) but still getting the above.

I have verified all my other 14+ jobs are running correctly, and they are all based on the same base set of scripts, the only thing that changes is the query. The other difference is, almost all the other scripts use an aggregation, but this one I am targetting a specific profile name. I have tested using an aggregation also, and same issue.

All connections use the AWS instance role to assume the correct role. e.g.:
```
[XXXX]
credential_source = Ec2InstanceMetadata
role_arn = arn:aws:iam::123:role/myrole
external_id = ZZZZZZZZZ
```

To summarize:
* Nothing changed in regards to the code I am using or our infrastructure in the last 2+ months
* Each invocation of the job executes in  a fresh container
* Have tried clearing the runner cache and completely simplify the query
* Each run installs latest aws-cli, aws plugin, csv plugin, steampipe + steampipe plugin
* Query works fine from my local machine

Let me know if any further logs/info is needed!

Potential related issues:
https://github.com/turbot/steampipe/issues/3482
https://github.com/turbot/steampipe/issues/3340

**Comment by captainfalcon23** [comment]
Update: It seems that my connection isn't being added to the search_path for some reason... ? 
```
connection "aws_blah" {
  plugin  = "aws"
  profile = "aupdns-blah"
  regions = ["ap-southeast-2"]
}

connection "aws_all" {
  plugin      = "aws"
  type        = "aggregator"
  connections = ["aws_*"]
}
```

```
su steampipe -c 'steampipe query "show search_path ;"'

+----------------------------+
| search_path                |
+----------------------------+
| public, steampipe_internal |
+----------------------------+
```

^ This is also using LATEST aws plugin and steampipe.

This all seems to work fine on my local machine though.

**Comment by captainfalcon23** [comment]
Update: I added:
```
sleep 1m
ps aux
su steampipe -c 'steampipe query "show search_path ;"'
```

before I ran my steampipe query, and also removed:
```
     - localedef -c -f UTF-8 -i en_US en_US.UTF-8
     - export LC_ALL=en_US.UTF-8
```
from my pipeline config (I have some small python script to send a message to slack), and it seems to work ok now. So, I think it is definitely related to this: https://github.com/turbot/steampipe/issues/3482#issuecomment-1564165630

Note that in the /home/steampipe/.steampipe/logs/database-XXX.log, it still logs the relation does not exist around 50 times, but then it magically starts working.... so I am very confused...

Note also the database-XXX.log is absolutely filled with thousands of these messages, unsure why these need to be logged here?

```
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."ena_support" is $steampipe_escape$Specifies whether enhanced networking with ENA is enabled.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."hypervisor" is $steampipe_escape$The hypervisor type of the instance. The value xen is used for both Xen and Nitro hypervisors.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."iam_instance_profile_arn" is $steampipe_escape$The Amazon Resource Name (ARN) of IAM instance profile associated with the instance, if applicable.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."iam_instance_profile_id" is $steampipe_escape$The ID of the instance profile associated with the instance, if applicable.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."image_id" is $steampipe_escape$The ID of the AMI used to launch the instance.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."instance_initiated_shutdown_behavior" is $steampipe_escape$Indicates whether an instance stops or terminates when you initiate shutdown from the instance (using the operating system command for system shutdown).$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."instance_lifecycle" is $steampipe_escape$Indicates whether this is a spot instance or a scheduled instance.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."kernel_id" is $steampipe_escape$The kernel ID$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."key_name" is $steampipe_escape$The name of the key pair, if this instance was launched with an associated key pair.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."launch_time" is $steampipe_escape$The time the instance was launched.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."outpost_arn" is $steampipe_escape$The Amazon Resource Name (ARN) of the Outpost, if applicable.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_affinity" is $steampipe_escape$The affinity setting for the instance on the Dedicated Host.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_availability_zone" is $steampipe_escape$The Availability Zone of the instance.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_group_id" is $steampipe_escape$The ID of the placement group that the instance is in.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_group_name" is $steampipe_escape$The name of the placement group the instance is in.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_host_id" is $steampipe_escape$The ID of the Dedicated Host on which the instance resides.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_host_resource_group_arn" is $steampipe_escape$The ARN of the host resource group in which to launch the instances.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_partition_number" is $steampipe_escape$The ARN of the host resource group in which to launch the instances.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."placement_tenancy" is $steampipe_escape$The tenancy of the instance (if the instance is running in a VPC). An instance with a tenancy of dedicated runs on single-tenant hardware.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."platform" is $steampipe_escape$The value is 'Windows' for Windows instances; otherwise blank.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."platform_details" is $steampipe_escape$The platform details value for the instance.$steampipe_escape$;
        COMMENT ON COLUMN "aws_some_connection"."aws_ec2_instance"."private_ip_address" is $steampipe_escape$The private IPv4 address assigned to the instance.$steampipe_escape$;

```

CC: @e-gineer

**Comment by captainfalcon23** [comment]
@kaidaguerre i ran the job today 10 times and it succeeded for all using the latest version. Let’s leave this issue open for a little longer and I will test it again next week and see. Still interesting it only affected this one table and no others.

**Comment by kaidaguerre** [comment]
that's great news

**Comment by captainfalcon23** [comment]
Hey @kaidaguerre 
So after the new release, and after my initial testing 3 days ago, out of all my jobs, 2 failed 
1. targeting the table aws_kms_keys but worked after a retry.
2. using CSV plugin, unable to read the csv generated by the previous step. I actually had to forcibly downgrade steampipe to v0.19.5 then it worked without any issues.

Let me know if you need any logs for either scenario (and what exactly) and I can share, but I think there is still something not quite right...

**Comment by captainfalcon23** [comment]
Hey @kaidaguerre , please see attached. Looks like there is some issue parsing the header:
```
2023-06-08 22:51:30.052 UTC [WARN]  failed to set connection config: failed to parse file header /builds/ops/reporting/reporting/aws-get-all-ec2-and-ami/aws-get-
all-ec2-and-ami_allamis.csv: EOF
```

[gitlab-logs.txt](https://github.com/turbot/steampipe/files/11696905/gitlab-logs.txt)

**Comment by captainfalcon23** [comment]
Haven't seen this one occur for a bit. Will close it for now and reopen if needed.
